src/javaharness/java/arcs/api/DeviceClientImpl.java

```diff
@@ -94,13 +87,10 @@ public void startArc(String json, Particle particle) {
     environment.sendMessageToArcs(jsonParser.stringify(request), null);
   }
 
-  protected void createPecForParticle(String pecid, Particle particle) {
-    if (!portById.containsKey(pecid)) {
-      portById.put(pecid, portFactory.createPECInnerPort(pecid, null));
-    }
-    PECInnerPort port = portById.get(pecid);
+  private void createPecForParticle(String pecId, Particle particle) {
```

Comment:
(noop - not related to this PR, and my fault, and probably should be fixed separately) I think, this method should only be available in the "remote" service, because it is meaningless to instantiate particle as non-factory inside Arcs service. 
I don't know if we can enforce this, but anyway, DeviceClient is a wrong place for it anyway - I mostly put it here because I needed to access the PecPort's map, but now that we have PortManager, this is not a problem anymore.

Thinking about it some more. `startArc` shouldn't be here either. But I guess, just ignore it for now, I'll fix it tomorrow.

Reply:
I'm not sure exactly where to put this code, but I'm all for you moving it somewhere else tomorrow :-)

That was the idea behind PecPortManager. If you need access to the PEC ports anywhere, just inject a PecPortManager. I'm sure its missing a ton of useful methods, we can add them in as they're needed.